### ISSUE TITLE
Error in src/app/product-details/product-details.component.html (6:16) : Object is possibly 'undefined'

### ISSUE THREAD

**Issue by ranjitkumartarlapalli** [angular/angular#43514]
### Description

In angular.io >> Try it >> Getting started, I opened the project in StackBlitz. The first section "Getting started" went through smoothly but in the next section "Adding navigation" the part "View product details" >> step 2, 3, 4, 5  I'm getting the error  "Error in src/app/product-details/product-details.component.html (6:16) : Object is possibly 'undefined' ".  Please help me

(https://angular.io/start/start-routing)

### What is the affected URL?

https://stackblitz.com/edit/angular-zl2udd?file=src%2Fapp%2Fproduct-details%2Fproduct-details.component.html

### Please provide the steps to reproduce the issue

In angular.io >> Try it >> Getting started, I opened the project in StackBlitz. The first section "Getting started" went through smoothly but in the next section "Adding navigation" the part "View product details" >> step 2, 3, 4, 5  I'm getting the error  "Error in src/app/product-details/product-details.component.html (6:16) : Object is possibly 'undefined' ".  Please help me

(https://angular.io/start/start-routing)

### Please provide a screenshot if possible

![Angular Object possibly undefined](https://user-images.githubusercontent.com/91046105/133988235-56b4856b-95f5-41a1-89c4-50fdf2a233fa.png)
![Angular Object possibly undefned](https://user-images.githubusercontent.com/91046105/133988407-341cf5e3-1fc4-46de-b967-d03524a2593b.png)

### Please provide the exception or error you saw

```true
Error in src/app/product-details/product-details.component.html (6:16) : Object is possibly 'undefined'
```

**Comment by atscott** [comment]
There needs to be a similar note to the one in the getting started step

> To make ProductAlertsComponent available to other components in the application, add it to AppModule's declarations in app.module.ts.

`ProductDetailsComponent` should be added to the `declarations` as well

**Comment by JoostK** [comment]
A fix is available in PR #43500.

**Comment by AndrewKushnir** [comment]
Closing this issue since PR #43500 was merged.
